### PR TITLE
added an object in images.tsx to hold urls of our static images 

### PR DESCRIPTION
**Pull request by CamilingJS**
Issue: #154 

#### WebP image formats offer superior compression, enhanced SEO benefits, and faster loading times. Including titles and descriptive alt text further improves SEO, as do descriptive file names. Additionally, consolidating static images within the images component simplifies the implementation of dynamic changes.

localhost_3000_

### CODE REVIEW

File: app/components/ui/images.tsx [hunk 1]
```diff
@@ -13,3 +13,113 @@ type ImageProps = ImgHTMLAttributes<HTMLImageElement> &
 export function Image({ ...props }: ImageProps) {
   return <UnpicImage {...props} />;
 }
+
+interface ImageDetails {
+  url: string;
+  altText: string;
+  title: string;
```
Comment (eiffelwong1) [inline app/components/ui/images.tsx:20]: do we need a required title field? wasn't able to see where it's used and felt a bit redundant from altText

File: app/components/marketing/events-section.tsx [hunk 2]
```diff
@@ -57,8 +58,9 @@ export function EventCard({ event, index }: { event: ExtendedEvent; index: numbe
         ) : (
           <img
```
Comment (eiffelwong1) [inline app/components/marketing/events-section.tsx:59]:
```suggestion
          <Image
```

Comment (CamilingJS) [reply app/components/marketing/events-section.tsx]: Done!

File: app/components/ui/images.tsx (hunk 1 above)
Comment (CamilingJS) [reply app/components/ui/images.tsx:20]: I've seen that the title field is primarily used to improve SEO ratings.

File: app/routes/about-us.tsx [hunk 3]
```diff
@@ -7,8 +7,9 @@ export default function AboutUs() {
         <div className="w-[320px] flex self-start mx-auto pb-3">
           <Image
             className="w-full"
-            src="https://res.cloudinary.com/dxctpvd8v/image/upload/v1706807323/solar-system_pzkazh.png"
-            alt="Solar system"
+            src={staticImage.happySun.url}
+            alt={staticImage.happySun.url}
```
Comment (andrelandgraf) [inline app/routes/about-us.tsx:11]:
```suggestion
            alt={staticImage.happySun.altText}
```

File: app/routes/about-us.tsx [hunk 4]
```diff
@@ -7,8 +7,9 @@ export default function AboutUs() {
...
+            title={staticImage.happySun.url}
```
Comment (andrelandgraf) [inline app/routes/about-us.tsx:12]:
```suggestion
            title={staticImage.happySun.title}
```

File: app/components/ui/images.tsx (hunk 1 above)
Comment (andrelandgraf) [reply app/components/ui/images.tsx:20]: Interesting! Do you have any blog posts or resources that you can share as a reference? Could be helpful to have this linked/documented in the PR description. Not a blocker at all, just curious!

File: app/components/marketing/hero-section.tsx [hunk 5]
```diff
@@ -58,15 +58,16 @@ export default function HeroSection() {
...
-            src="/imgs/Hero-Image.png"
-            alt="group of people"
+            src={staticImage.largeGroup.url}
+            alt={staticImage.largeGroup.altText}
+            title={staticImage.largeGroup.title}
             height={379}
             width={553}
             priority
-          />
+          ></Image>
```
Comment (andrelandgraf) [inline app/components/marketing/hero-section.tsx:70]: Is this change intended/needed?

```suggestion
          />
```

**Review by eiffelwong1 — COMMENTED** [review]
LGTM, will approve once minor issue is changed. :D

File: app/components/ui/images.tsx (hunk 1 above)
Comment (CamilingJS) [reply app/components/ui/images.tsx]: title will be optional!

File: app/components/marketing/events-section.tsx [hunk 6]
```diff
@@ -44,18 +45,23 @@ export function EventCard({ event, index }: { event: FullEvent; index: number })
...
       <div className="flex flex-col space-y-3 w-64 md:h-96 md:w-72 mx-6 sm:mx-0 max-w-md p-6 rounded-2xl bg-primary hover:transform hover:scale-105 transition-transform duration-300 hover:border">
         {event.imgUrl && event.imgAlt ? (
-          <img
+          <Image
             className={`${index === 1 && 'sm:order-2 pt-8'}  w-full object-contain box-border`}
             src={event.imgUrl}
             alt={event.imgAlt}
             style={{ viewTransitionName: isTransitioning ? 'image-expand' : '' }}
+            width={300}
+            height={200}
```
Comment (shami-dev) [inline app/components/marketing/events-section.tsx:54]: Same here!

File: app/components/marketing/events-section.tsx [hunk 7]
```diff
@@ -44,18 +45,23 @@ export function EventCard({ event, index }: { event: FullEvent; index: number })
...
+          <Image
             className={`${index === 1 && 'sm:order-2 pt-8'}  w-full object-contain box-border`}
-            src={'https://res.cloudinary.com/dxctpvd8v/image/upload/v1709188683/default-event-photo_bvdslj.png'}
-            alt={'Default animated event zoom meeting'}
+            src={staticImage.defaultEventPhoto.url}
+            alt={staticImage.defaultEventPhoto.altText}
+            title={staticImage.defaultEventPhoto.title}
             style={{ viewTransitionName: isTransitioning ? 'image-expand' : '' }}
+            width={300}
+            height={200}
```
Comment (shami-dev) [inline app/components/marketing/events-section.tsx:64]: And here!

File: app/components/marketing/events-section.tsx [hunk 8]
```diff
@@ -44,18 +45,23 @@ export function EventCard({ event, index }: { event: FullEvent; index: number })
...
           />
         ) : (
-          <img
+          <Image
             className={`${index === 1 && 'sm:order-2 pt-8'}  w-full object-contain box-border`}
-            src={'https://res.cloudinary.com/dxctpvd8v/image/upload/v1709188683/default-event-photo_bvdslj.png'}
-            alt={'Default animated event zoom meeting'}
+            src={staticImage.defaultEventPhoto.url}
+            alt={staticImage.defaultEventPhoto.altText}
+            title={staticImage.defaultEventPhoto.title}
```
Comment (shami-dev) [inline app/components/marketing/events-section.tsx:61]: Generally, using the title attribute is not recommended (see my research in Discord). However, it can be beneficial for SEO for the marketing pages.

**Review by eiffelwong1 — DISMISSED** [review]
LGTM :D

File: app/components/ui/images.tsx [hunk 9]
```diff
@@ -27,3 +27,113 @@ export function Image({ ...props }: ImageProps) {
...
+
+interface ImageDetails {
+  url: string;
+  altText: string;
+  title?: string;
+}
+
+const staticImage: { [key: string]: ImageDetails } = {
+  avatarAstronaut: {
+    url: 'https://res.cloudinary.com/dxctpvd8v/image/upload/v1716394728/SocialPlanIt/avatar-astronaut_100x100.webp',
```
Comment (andrelandgraf) [inline app/components/ui/images.tsx:39]: I think we could even remove the `web` file ending, and Cloudinary would just use the best possible format. Not a blocker, but maybe something to experiment with!

**Review by andrelandgraf — APPROVED** [review]
LGTM - feel free to make a decision regarding the `title` prop (if we make it optional or remove it or keep it). Thanks for the great work!

**Review by eiffelwong1 — APPROVED** [review]
lgtm
